crates/oxc_ast/src/ast/js.rs

```diff
@@ -1300,7 +1300,7 @@ pub struct TryStatement<'a> {
 }
 
 #[visited_node]
-#[scope(if(self.param.is_some()))]
```

Comment:
@Dunqing What was reason for removing `if(self.param.is_some())`? If no catch param, we don't need a scope, right?

I think it's arguable that it would be simpler if we didn't have any conditional scopes, but if we do that, probably we should do it consistently.

Reply:
Yes, just to be consistent with the spec

Reply:
OK, so is plan to make all scopes unconditional?

Reply:
Personally, I don't think we *need* to adhere to the spec. The spec defines what *observable behavior* must be, but implementations are free to achieve that through any means they choose. V8, for example, diverges from the example algorithms in the spec very heavily for purpose of better performance.

Reply:
I agree. I have found that there is a performance impact of `boa` reporting errors according to the specification

https://github.com/boa-dev/boa/blob/5b1621abd2b702b99bd6b4491973095f9b27c362/core/parser/src/parser/statement/try_stm/catch.rs#L92-L97

Reply:
Thanks for coming back. OK. I would suggest reverting this to `#[scope(if(self.param.is_some()))]` for now then. If we keep scopes as conditional, it's quite easy to make them all unconditional later. But it's harder to go in other direction - you have to go through all the scopes and reason about which *could* be conditional.

Personally I think the most important thing either way is to be consistent, as it's one less oddity to reason about.

But that's just my opinion. You can disagree, and I won't argue further.

Reply:
Ok. I will send a PR later